feature_engine/selection/probe_feature_selector.py

```diff
@@ -0,0 +1,233 @@
+    _initial_model_performance_docstring,
+    _scoring_docstring,
+    _threshold_docstring,
+    _transform_docstring,
+    _variables_attribute_docstring,
+    _variables_numerical_docstring,
+)
+
+class ProbeFeatureSelection(BaseSelector):
+    """
+
+
+    """
+    def __init__(
+        self,
+        estimator,
+        scoring: str = "roc_auc",
+        distribution: str = "normal",
+        cut_rate: float = 0.2,
+        cv: int = 5,
+        n_iter: int = 5,
+        random_state: int = 0,
+        # TODO: Do we need confirm_variable given that this selector will not be used in a pipeline?
+        # TODO: Do we need the parameter because it is a param of BaseSelector?
+        confirm_variables: bool = False,
+    ):
+
+        if distribution not in ["normal", "binary", "unfirom"]:
+            raise ValueError(
+                "distribution takes on normal, binary, or uniform as values. "
+                f"Got {distribution} instead."
+            )
+
+        if not (0 <= cut_rate <= 1):
+            raise ValueError(
+                "cut_rate must range from 0 to 1, inclusively. "
+                f"Got {cut_rate} instead."
+            )
+
+        if not isinstance(cv, int) or not (cv > 0):
+            raise ValueError(
+                f"cv must be a positive integer. Got {cv} instead."
+            )
+
+        if not isinstance(n_iter, int) or not (n_iter > 0):
+            raise ValueError(
+                f"n_iter must be a positive integer. Got {n_iter} instead."
+            )
+        super().__init__(confirm_variables)
+        self.estimator = estimator
+        self.scoring = scoring
+        self.distribution = distribution
+        self.cut_rate = cut_rate
+        self.cv = cv
+        self.n_iter = n_iter
+        self.random_state = random_state
+
+    def fit(self, X: pd.DataFrame, y: pd.Series):
+        """
+        Create three random feature. Find initial model performance.
+        Sort features by importance.
+
+        Parameters
+        ----------
+        X: pandas dataframe of shape = [n_samples, n_features]
+           The input dataframe
+        y: array-like of shape (n_samples)
+           Target variable. Required to train the estimator.
+        """
+        # check input dataframe
+        X, y = check_X_y(X, y)
+
+        # find numerical variables
+        self.variables = _find_or_check_numerical_variables(X, None)
+
+        # if required exclude variables that are not in the input dataframe
+        # instantiates the 'variables_' attribute
+        self._confirm_variables(X)
+
+        # check that there is more than 1 variable
+        self._check_variable_number()
+
+        # save input features
+        self._get_feature_names_in(X)
+
+        # create probe feature distribution
+        self.probe_feature_data_ = self._generate_probe_feature(X.shape[0])
+
+        # get probe feature name
+        self.probe_feature_ = self.probe_feature_data_.columns
```

Comment:
Let's unify the 2 last parameters into one, called `probe_features_` that will contain a dataframe or a dictionary with the probe features that were created for the assessment.